pkg/server/service_mock.go

```diff
@@ -32,14 +41,39 @@ import (
 
 // A testing mock that wraps server.Server to Start and defer Stop a server
 type MockServer struct {
-	gc *GRPCConfig
-	hc *HTTPConfig
-	wg *sync.WaitGroup
+	gc           *GRPCConfig
+	hc           *HTTPConfig
+	wg           *sync.WaitGroup
+	cleanupFiles []string
 }
 
-func (ms *MockServer) Start(_ *testing.T) {
-	ms.gc = NewGRPCConfig()
-	ms.hc = NewHTTPConfig()
+func (ms *MockServer) Start(t *testing.T, enableTLS bool) {
```

Comment:
Can we add a `StartTLS` method instead of updating `Start`? It's roughly the same amount of code and makes it more obvious to developers that they're initializing a server with or without TLS.